src/praisonai/praisonai/gateway/auth.py

```
@@ -0,0 +1,215 @@
+            existing_lines.append(f"GATEWAY_AUTH_TOKEN={token}\n")
+        
+        # Write back to file with secure permissions
+        with open(env_file_path, 'w') as f:
+            f.writelines(existing_lines)
+        
+        # Set secure file permissions (owner read/write only)
+        os.chmod(env_file_path, 0o600)
+        
+        logger.debug(f"Gateway auth token persisted to {env_file_path}")
+        
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)